platform/nephos/sonic-platform-modules-cig/cs6436-56p/modules/x86-64-cig-cs6436-56p-sfp.c

```diff
@@ -0,0 +1,1454 @@
+cs6436_56p_sfp29, cs6436_56p_sfp30, cs6436_56p_sfp31, cs6436_56p_sfp32,
+cs6436_56p_sfp33, cs6436_56p_sfp34, cs6436_56p_sfp35, cs6436_56p_sfp36,
+cs6436_56p_sfp37, cs6436_56p_sfp38, cs6436_56p_sfp39, cs6436_56p_sfp40,
+cs6436_56p_sfp41, cs6436_56p_sfp42, cs6436_56p_sfp43, cs6436_56p_sfp44,
+cs6436_56p_sfp45, cs6436_56p_sfp46, cs6436_56p_sfp47, cs6436_56p_sfp48,
+cs6436_56p_sfp49, cs6436_56p_sfp50, cs6436_56p_sfp51, cs6436_56p_sfp52,
+cs6436_56p_sfp53, cs6436_56p_sfp54, cs6436_56p_sfp55, cs6436_56p_sfp56
+};
+
+#define I2C_DEV_ID(x) { #x, x}
+
+static const struct i2c_device_id sfp_device_id[] = {
+I2C_DEV_ID(cs6436_56p_sfp1),
+I2C_DEV_ID(cs6436_56p_sfp2),
+I2C_DEV_ID(cs6436_56p_sfp3),
+I2C_DEV_ID(cs6436_56p_sfp4),
+I2C_DEV_ID(cs6436_56p_sfp5),
+I2C_DEV_ID(cs6436_56p_sfp6),
+I2C_DEV_ID(cs6436_56p_sfp7),
+I2C_DEV_ID(cs6436_56p_sfp8),
+I2C_DEV_ID(cs6436_56p_sfp9),
+I2C_DEV_ID(cs6436_56p_sfp10),
+I2C_DEV_ID(cs6436_56p_sfp11),
+I2C_DEV_ID(cs6436_56p_sfp12),
+I2C_DEV_ID(cs6436_56p_sfp13),
+I2C_DEV_ID(cs6436_56p_sfp14),
+I2C_DEV_ID(cs6436_56p_sfp15),
+I2C_DEV_ID(cs6436_56p_sfp16),
+I2C_DEV_ID(cs6436_56p_sfp17),
+I2C_DEV_ID(cs6436_56p_sfp18),
+I2C_DEV_ID(cs6436_56p_sfp19),
+I2C_DEV_ID(cs6436_56p_sfp20),
+I2C_DEV_ID(cs6436_56p_sfp21),
+I2C_DEV_ID(cs6436_56p_sfp22),
+I2C_DEV_ID(cs6436_56p_sfp23),
+I2C_DEV_ID(cs6436_56p_sfp24),
+I2C_DEV_ID(cs6436_56p_sfp25),
+I2C_DEV_ID(cs6436_56p_sfp26),
+I2C_DEV_ID(cs6436_56p_sfp27),
+I2C_DEV_ID(cs6436_56p_sfp28),
+I2C_DEV_ID(cs6436_56p_sfp29),
+I2C_DEV_ID(cs6436_56p_sfp30),
+I2C_DEV_ID(cs6436_56p_sfp31),
+I2C_DEV_ID(cs6436_56p_sfp32),
+I2C_DEV_ID(cs6436_56p_sfp33),
+I2C_DEV_ID(cs6436_56p_sfp34),
+I2C_DEV_ID(cs6436_56p_sfp35),
+I2C_DEV_ID(cs6436_56p_sfp36),
+I2C_DEV_ID(cs6436_56p_sfp37),
+I2C_DEV_ID(cs6436_56p_sfp38),
+I2C_DEV_ID(cs6436_56p_sfp39),
+I2C_DEV_ID(cs6436_56p_sfp40),
+I2C_DEV_ID(cs6436_56p_sfp41),
+I2C_DEV_ID(cs6436_56p_sfp42),
+I2C_DEV_ID(cs6436_56p_sfp43),
+I2C_DEV_ID(cs6436_56p_sfp44),
+I2C_DEV_ID(cs6436_56p_sfp45),
+I2C_DEV_ID(cs6436_56p_sfp46),
+I2C_DEV_ID(cs6436_56p_sfp47),
+I2C_DEV_ID(cs6436_56p_sfp48),
+I2C_DEV_ID(cs6436_56p_sfp49),
+I2C_DEV_ID(cs6436_56p_sfp50),
+I2C_DEV_ID(cs6436_56p_sfp51),
+I2C_DEV_ID(cs6436_56p_sfp52),
+I2C_DEV_ID(cs6436_56p_sfp53),
+I2C_DEV_ID(cs6436_56p_sfp54),
+I2C_DEV_ID(cs6436_56p_sfp55),
+I2C_DEV_ID(cs6436_56p_sfp56),
+{ /* LIST END */ }
+};
+MODULE_DEVICE_TABLE(i2c, sfp_device_id);
+
+/*
+ * list of valid port types
+ * note OOM_PORT_TYPE_NOT_PRESENT to indicate no
+ * module is present in this port
+ */
+typedef enum oom_driver_port_type_e {
+	OOM_DRIVER_PORT_TYPE_INVALID,
+	OOM_DRIVER_PORT_TYPE_NOT_PRESENT,
+	OOM_DRIVER_PORT_TYPE_SFP,
+	OOM_DRIVER_PORT_TYPE_SFP_PLUS,
+	OOM_DRIVER_PORT_TYPE_QSFP,
+	OOM_DRIVER_PORT_TYPE_QSFP_PLUS,
+	OOM_DRIVER_PORT_TYPE_QSFP28
+} oom_driver_port_type_t;
+
+enum driver_type_e {
+	DRIVER_TYPE_SFP_MSA,
+	DRIVER_TYPE_SFP_DDM,
+	DRIVER_TYPE_QSFP
+};
+
+/* Each client has this additional data
+ */
+struct eeprom_data {

```

Comment:
indentation for these structs looked odd, is it possible to fix them?

Reply:
The code is similar as the Linux optoe driver "drivers/misc/eeprom/eeprom.c": 

/* Each client has this additional data */
struct eeprom_data {
        struct mutex update_lock;
        u8 valid;                       /* bitfield, bit!=0 if slice is valid */
        unsigned long last_updated[8];  /* In jiffies, 8 slices */
        u8 data[EEPROM_SIZE];           /* Register values */
        enum eeprom_nature nature;
};